src/actions/sources/loadSourceText.js

```diff
@@ -82,20 +82,24 @@ export function loadSourceText(source: SourceRecord) {
     }
 
     requests.set(id, deferred.promise);
-    await dispatch({
-      type: "LOAD_SOURCE_TEXT",
-      sourceId: id,
-      [PROMISE]: loadSource(source, { sourceMaps, client })
-    });
+    try {
+      await dispatch({
+        type: "LOAD_SOURCE_TEXT",
+        sourceId: id,
+        [PROMISE]: loadSource(source, { sourceMaps, client })
+      });
+    } catch (e) {
+      // Ignoring the rejected promise -- source already has 'error' property.
+    }
```

Comment:
can we move the catch into `loadSource` so that it doesn't reject the promise?

Reply:
Nope, you need dispatch() to send the "error" status.

Reply:
i see...